util/template/expression_template.go

```diff
@@ -80,9 +81,20 @@ func expressionReplace(w io.Writer, expression string, env map[string]interface{
 	if resultMarshaled == nil {
 		return 0, fmt.Errorf("failed to marshal evaluated marshaled expression %q", expression)
 	}
-	// Trim leading and trailing quotes. The value is being inserted into something that's already a string.
 	marshaledLength := len(resultMarshaled)
-	return w.Write(resultMarshaled[1 : marshaledLength-1])
+
+	// Trim leading and trailing quotes. The value is being inserted into something that's already a string.
+	if len(resultMarshaled) > 1 && resultMarshaled[0] == '"' && resultMarshaled[marshaledLength-1] == '"' {
+		return w.Write(resultMarshaled[1 : marshaledLength-1])
+	}
+
+	// Check if the value contains any quotes by adding double quotes at the beginning and end of the value.
+	// If so then escape quotes.
```

Comment:
Why are we escaping quotes here? Shouldn't the user should be free to choose whether they want to do this or not?

Reply:
As you can see from the code line below, we're checking if we surround the result value with quotes is a valid JSON or not. The reason is, the value could be an array or object and if it's invalid(and you wouldn't want to return an invalid JSON value) when you surround it with quotes then it means it contains quotes inside of it already, so we escape it(and you wouldn't want to return an invalid JSON value).

Here is an example what it returns with these new changes and current code:

You can `Replace` function `util/template` mod with this input:
`{{=jsonpath('{"employees": [{"name": "Baris", "age":43}, {"name": "Mo", "age": 42}]}', '$.employees[*].name')}}`

With this new code, it returns: `{"value":"[\"Baris\",\"Mo\"]"}` which is a valid value.
And with current code, it returns: `{"value":""Baris","Mo""}` as you can see this is an invalid value.

You can experiment with it enabling/disabling my changes here and by modifying/tweaking test cases I added in `template_test.go` file.

I hope this answers your question.

Reply:
Yes, this does.
I was wondering if it would just make sense to simplify it down to always just quote.

```
    resultQuoted := []byte(strconv.Quote(string(resultMarshaled)))
	return w.Write(resultQuoted[1 : len(resultQuoted)-1])
```

The json.Valid test is probably more expensive than just quoting always.

As it is the code is hard to comprehend why anyone might need to check that json.Marshal returns valid json.

Reply:
Actually, you're right. We don't need to check if the result is correct json or not, we can always quote. I did the change. Thanks!